.github/workflows/publish.yml

```diff
@@ -21,18 +26,23 @@ jobs:
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 
-      # Build before generating build number so the build number doesn't needlessly increase in the event of an error.
-      - name: Build
-        run: ./gradlew build javadocJar --stacktrace
-      # Generate the build number based on tags to allow per branch build numbers, not something GitHub provides by default.
-      - name: Generate build number
-        id: buildnumber
-        uses: onyxmueller/build-tag-number@v1
-        with:
-          token: ${{ secrets.github_token }}
-          prefix: "build/${{ github.ref }}"
+      # Fetch tags because they're used to store build number.
+      - name: Fetch tags
+        run: git fetch --tags origin
+      # Check for existing tag created by previous runs of "Increment build number".
+      # Extract build number if present, otherwise start at 0.
+      - name: Set build number
+        shell: bash
+        run: |
+          output=$(git tag -l build/${{ github.ref }}-build-number-*)
+          buildNumber=$(if [[ $output ]]; then echo $output | grep -oP "(?<=build-number-)[0-9]+$"; else echo 0; fi)
```

Comment:
Note that this assumes only one tag was found; this should be true so long as we don't set `delete_previous_tag: false` for `onyxmueller/build-tag-number@v1`.